### PR TITLE
Hotfix missing discussion items url

### PR DESCRIPTION
**Pull request by Yago**
I'm aware it's not super clean, but it does the trick until Github fix its REST API (see #424). If so, the first test will pass and this code will never reach out again and we will be able to remove it without second thought 😉

Again, it's just an idea for the mean time, feel free to say that it's a bad idea. I didn't update the test due to the very “temporary” aspect of this code, but I can do so if needed.

### CODE REVIEW

File: src/components/NotificationRow.tsx [hunk 1]
```diff
@@ -35,6 +35,18 @@ export const NotificationRow: React.FC<IProps> = ({
       const url = generateGitHubWebUrl(notification.subject.url);
       shell.openExternal(url);
     }
+
```
Comment (codebytere) [inline src/components/NotificationRow.tsx:38]: We know the first and third conditionals will always be true at this point of code execution, so we probably can clean this up a little bit:

```js
    // Some Notification types from GitHub are missing urls in their subjects.
    if (notification.subject.url) {
      const url = generateGitHubWebUrl(notification.subject.url);
      shell.openExternal(url);
    // For discussions, we can at least send users to the main discussions page.
    } else if (notification.subject.type === SubjectType.Discussion) {
      const url = generateGitHubWebUrl(`${notification.repository.url}/discussions`);
      shell.openExternal(url);
    }
```

**Review by codebytere — COMMENTED** [review]
Thanks! Looking good, just a cleanup comment.

**Comment by setchy** [comment]
Looking forward to this enhancement.  Thanks @Yago for contributing this PR

File: src/components/NotificationRow.tsx (hunk 1 above)
Comment (Yago) [reply src/components/NotificationRow.tsx]: Done 😉

**Comment by setchy** [comment]
@manosim - do you think this enhancement is ready to merge and release?

**Comment by Araxeus** [comment]
@manosim if this really fix the issue, why not merge this?

Its really annoying that discussions notifications straight up don't work

**Comment by Araxeus** [comment]
we can get the actual Discussion url using GraphQL

I have created a repo to easily test this: https://github.com/Araxeus/github_discussion_notification_url

Any thoughts?

**Comment by afonsojramos** [comment]
Will merge #538 over this one.

**Pull request closed without merging**